src/pipeline_manager.py

```diff
@@ -135,19 +166,23 @@ def predict(pipeline_name, dev_mode, submit_predictions, chunk_size):
 
 def visualize(pipeline_name, image_dir=None, single_image=None, n_files=16, show_popups=False):
     if image_dir:
-        SOLUTION_CONFIG['loader']['dataset_params']['images_dir'] = image_dir
         img_ids = get_img_ids_from_folder(SOLUTION_CONFIG['loader']['dataset_params']['images_dir'], n_ids=n_files)
+        metadata = generate_metadata(valid_image_ids=img_ids, valid_image_dir=image_dir)
+        _, _, _, images_data = _get_input_data(metadata=metadata)
```

Comment:
I know I am also a culprit of doing this _, _ thing but it feels like this is a spot to use extended unpacking:

```python
a = (1,2,3,4,5)
_ , *b = a
```